### PR TITLE
Avoid NPE as discussed in #57

### PR DESCRIPTION
**Pull request by TheSnoozer**
revWalk.parseCommit will throw a NPE when Argument is null.

```
Caused by: java.lang.NullPointerException
	at org.eclipse.jgit.lib.ObjectIdOwnerMap.get(ObjectIdOwnerMap.java:131)
	at org.eclipse.jgit.revwalk.RevWalk.parseAny(RevWalk.java:838)
	at org.eclipse.jgit.revwalk.RevWalk.parseCommit(RevWalk.java:753)
	at pl.project13.maven.git.JGitProvider.prepareGitToExtractMoreDetailedReproInformation(JGitProvider.java:91)
	... 26 more
```

### CODE REVIEW

File: src/main/java/pl/project13/maven/git/JGitProvider.java [hunk 1]
```diff
@@ -83,12 +83,16 @@ protected String getBuildAuthorEmail() {
...
-      if (HEAD == null) {
+      Ref head = git.getRef(Constants.HEAD);
+      if (head == null) {
         throw new MojoExecutionException("Could not get HEAD Ref, are you sure you've set the dotGitDirectory property of this plugin to a valid path?");
       }
       revWalk = new RevWalk(git);
-      headCommit = revWalk.parseCommit(HEAD.getObjectId());
+      ObjectId headObjectId = head.getObjectId();
+      if(headObjectId == null){
+        throw new MojoExecutionException("Could not get HEAD Ref, are you sure you've some commits in the dotGitDirectory?");
```
Comment (ktoso) [inline src/main/java/pl/project13/maven/git/JGitProvider.java]: you have

**Comment by ktoso** [comment]
Good one, [LGTM](http://www.urbandictionary.com/define.php?term=LGTM&defid=2535452), I'd reword the msg though :-)

**Comment by TheSnoozer** [comment]
Changed to you have. Just copied and pasted from the other message by the way ;)

**Comment by ktoso** [comment]
oh ok, thanks :) let's merge
